drivers/net/bnxt/bnxt_hwrm.c

```diff
@@ -1530,3 +1538,81 @@ int bnxt_hwrm_func_qcfg(struct bnxt *bp)
 
 	return rc;
 }
+
+int bnxt_hwrm_allocate_vfs(struct bnxt *bp, int num_vfs)
+{
+	struct hwrm_func_cfg_input req = {0};
+	struct hwrm_func_cfg_output *resp = bp->hwrm_cmd_resp_addr;
+	struct hwrm_func_qcfg_input qreq = {0};
+	struct hwrm_func_qcfg_output *qresp = bp->hwrm_cmd_resp_addr;
+	int i;
+	int rc = 0;
+
+	if (!BNXT_PF(bp)) {
+		RTE_LOG(ERR, PMD, "Attempt to allcoate VFs on a VF!\n");
+		return -1;
+	}
+
+	bp->pf.active_vfs = 0;
+
+	req.enables = rte_cpu_to_le_32(HWRM_FUNC_CFG_INPUT_ENABLES_MTU |
+			HWRM_FUNC_CFG_INPUT_ENABLES_MRU |
+			HWRM_FUNC_CFG_INPUT_ENABLES_NUM_RSSCOS_CTXS |
+			HWRM_FUNC_CFG_INPUT_ENABLES_NUM_STAT_CTXS |
+			HWRM_FUNC_CFG_INPUT_ENABLES_NUM_CMPL_RINGS |
+			HWRM_FUNC_CFG_INPUT_ENABLES_NUM_TX_RINGS |
+			HWRM_FUNC_CFG_INPUT_ENABLES_NUM_RX_RINGS |
+			HWRM_FUNC_CFG_INPUT_ENABLES_NUM_L2_CTXS |
+			HWRM_FUNC_CFG_INPUT_ENABLES_NUM_VNICS |
+			HWRM_FUNC_CFG_INPUT_ENABLES_NUM_HW_RING_GRPS);
+
+	req.mtu = rte_cpu_to_le_16(bp->eth_dev->data->mtu + ETHER_HDR_LEN + ETHER_CRC_LEN + VLAN_TAG_SIZE);
+	req.mru = rte_cpu_to_le_16(bp->eth_dev->data->mtu + ETHER_HDR_LEN + ETHER_CRC_LEN + VLAN_TAG_SIZE);
+	req.num_rsscos_ctxs = rte_cpu_to_le_16(bp->pf.max_rsscos_ctx / (num_vfs + 1));
+	req.num_stat_ctxs = rte_cpu_to_le_16(bp->pf.max_stat_ctx / (num_vfs + 1));
+	req.num_cmpl_rings = rte_cpu_to_le_16(bp->pf.max_cp_rings / (num_vfs + 1));
+	req.num_tx_rings = rte_cpu_to_le_16(bp->pf.max_tx_rings / (num_vfs + 1));
+	req.num_rx_rings = rte_cpu_to_le_16(bp->pf.max_rx_rings / (num_vfs + 1));
+	req.num_l2_ctxs = rte_cpu_to_le_16(bp->pf.max_l2_ctx / (num_vfs + 1));
+	req.num_vnics = rte_cpu_to_le_16(bp->pf.max_vnics / (num_vfs + 1));
+	req.num_hw_ring_grps = rte_cpu_to_le_16(bp->max_ring_grps / (num_vfs + 1));
+
+	for (i = 0; i < num_vfs; i++) {
+		req.enables &= ~rte_le_to_cpu_32(HWRM_FUNC_CFG_INPUT_ENABLES_DFLT_MAC_ADDR);
+		/* Check for zero MAC address */
```

Comment:
If I am seeing it correctly, qreq and qresp are used only in this scope and could be declared locally.